src/org/zowe/jenkins_shared_library/scm/GitHub.groovy

```diff
@@ -1018,4 +1039,56 @@ class GitHub {
         this.steps.echo "User $user is "+(isUserWrite ? "" : "not ")+"a permitted write collaborator"
         return isUserWrite
     }
+
+    /**
+     * Post a comment on issue/pr
+     *
+     * @param  issueNum     issue or pr number
+     * @return              comment posted timestamp
+     */
+    String postComment(Integer issueNum, String contentString) {
+        // Note: in Github APIs, PR is a subset of issue, but with code changes
+        //       issueNum can be either issue number or PR number
+
+        // validate arguments
+        validateArgs()
+
+        //additional validation
+        if (!issueNum) {
+            throw new InvalidArgumentException('issueNum')
+        }
+        if (!contentString) {
+            throw new InvalidArgumentException('contentString')
+        }
+
+        def result
+
+        this.steps.withCredentials([
+            this.steps.usernamePassword(
+                credentialsId: this.usernamePasswordCredential,
+                passwordVariable: 'PASSWORD',
+                usernameVariable: 'USERNAME'
+            )
+        ]) {
+            this.steps.echo "Posting a comment on issue/pr $issueNum on ${this.repository} \n content is $contentString"
+            def cmd_postComment = "curl -u \"\${USERNAME}:\${PASSWORD}\" -sS" +
+                    " -X POST" +
+                    " -H \"Accept: application/vnd.github.v3+json\"" +
+                    " \"https://${GITHUB_API_DOMAIN}/repos/${this.repository}/issues/$issueNum/comments\"" +
+                    " -d '{\"body\":\"$contentString\"}'"
```

Comment:
I think the only thing we need to careful is the contentString when it includes quotations.